mypy/expandtype.py

```diff
@@ -98,7 +99,28 @@ def visit_type_var(self, t: TypeVarType) -> Type:
         else:
             return repl
 
+    def visit_param_spec(self, t: ParamSpecType) -> Type:
+        repl = get_proper_type(self.variables.get(t.id, t))
+        if isinstance(repl, Instance):
+            inst = repl
+            # Return copy of instance with type erasure flag on.
+            return Instance(inst.type, inst.args, line=inst.line,
+                            column=inst.column, erased=True)
+        else:
+            return repl
+
     def visit_callable_type(self, t: CallableType) -> Type:
+        param_spec = t.param_spec()
+        if param_spec is not None:
+            repl = get_proper_type(self.variables.get(param_spec.id))
+            if isinstance(repl, CallableType):
```

Comment:
TBH it is hard to follow the logic here: could you please add a comment on why do we need to have some non-trivial logic for callable?

Reply:
Added a comment.